src/Compilers/CSharp/Portable/Symbols/NativeIntegerTypeSymbol.cs

```diff
@@ -35,22 +40,29 @@ internal NativeIntegerTypeSymbol(NamedTypeSymbol underlying) : base(underlying,
 
         public override SpecialType SpecialType => _underlyingType.SpecialType;
 
-        public override IEnumerable<string> MemberNames => Array.Empty<string>();
+        public override IEnumerable<string> MemberNames => GetMembers().SelectAsArray(m => m.Name);
 
         /// <summary>
-        /// There are no members on <see cref="System.IntPtr"/> that should be exposed directly on nint:
+        /// There only members of <see cref="System.IntPtr"/> that are exposed directly on nint
+        /// are the overridden methods Equals(), GetHashCode(), and ToString(). For the other members:
         /// constructors for nint other than the default parameterless constructor are not supported;
         /// operators are handled explicitly as built-in operators and conversions;
         /// 0 should be used instead of Zero;
         /// sizeof(nint) should be used instead of Size;
         /// + and - should be used instead of Add() and Subtract();
-        /// ToInt32(), ToInt64(), ToPointer() should be used from System.IntPtr only;
-        /// overridden methods Equals(), GetHashCode(), and ToString() are referenced from System.Object.
```

Comment:
>/// overridden methods Equals(), GetHashCode(), and ToString() are referenced from System.Object. [](start = 8, length = 97)

Just curious, why doing this wasn't good enough?

Reply:
It resulted in bad code such as:
```
class C
{
    readonly nint n;
    string F() => n.ToString();
}
```
Result:
```
  IL_0000:  ldarg.0
  IL_0001:  ldflda     native int C::n
  IL_0006:  constrained. [mscorlib]System.IntPtr
  IL_000c:  callvirt   instance string [mscorlib]System.Object::ToString()
  IL_0011:  ret
```
Expected:
```
  .locals init (native int V_0)
  IL_0000:  ldarg.0
  IL_0001:  ldfld      native int C::n
  IL_0006:  stloc.0
  IL_0007:  ldloca.s   V_0
  IL_0009:  call       instance string [mscorlib]System.IntPtr::ToString()
  IL_000e:  ret
```
See `ReadOnlyField_VirtualMethods` test.

---
In reply to: [382292681](https://github.com/dotnet/roslyn/pull/41728#discussion_r382292681) [](ancestors = 382292681)

Reply:
> It resulted in bad code generated for callers 

Let's imaging there is a virtual method in base that IntPtr doesn't override. Would we generate the same bad code for that method invocation?

---
In reply to: [382302890](https://github.com/dotnet/roslyn/pull/41728#discussion_r382302890) [](ancestors = 382302890,382292681)

Reply:
It sounds like there is a bug in the verifier. I can build and successfully run this code:
```
class C
{
    readonly nint n1;
    readonly System.IntPtr n2;
    string F1() => n1.ToString();
    string F2() => n2.ToString();

    static void Main()
    {
        System.Console.WriteLine(new C().F1());
        System.Console.WriteLine(new C().F2());
    }
}

struct nint
{}
```

But PEVerify complains about IL in F1.

So, if that verification error is the only reason to synthesize members, I think we should give this another thought.


---
In reply to: [382316178](https://github.com/dotnet/roslyn/pull/41728#discussion_r382316178) [](ancestors = 382316178,382302890,382292681)

Reply:
I'll remove the members from `NativeIntegerTypeSymbol` for now.

---
In reply to: [382719607](https://github.com/dotnet/roslyn/pull/41728#discussion_r382719607) [](ancestors = 382719607,382316178,382302890,382292681)